### ISSUE TITLE
Error in proxy_ldap mode

### ISSUE THREAD

**Issue by Mystikal57** [pulsejet/nextcloud-oidc-login#52]
Hello,

I added LDAP_ID from keycloak but i get this error after:
Error getting DN for LDAP user

thank you

**Comment by pulsejet** [comment]
Is your LDAP configuration correct? Can you login with normal LDAP-bind
(username/password)?

**Comment by Mystikal57** [comment]
Yes i can login with normal ldap way
i synced all my users in nextcloud

**Comment by pulsejet** [comment]
The uid and ldapUid properties are configurable in `config.php`; it's the responsibility of the oidc provider to return the correct value here

**Comment by Mystikal57** [comment]
yes i modified when i see UID, i search for ldap guid like in nextcloud.
You should add it to readme (ldapuid = samacountname / username)

**Comment by pulsejet** [comment]
> `$uid = strtoupper($profile[$attr['id']]);`

I can't do this as it breaks compatibility for existing setups using OpenLDAP. If you need uppercase UID, the "id" attribute supplied by the OIDC client should be in uppercase (or patch this plugin). That said, not sure what the `preg_replace` is doing anyway (it isn't even valid regex), so that'll have to go.

**Comment by andreisilvasantos** [comment]
Hello. I am using Keycloak as an OID Provider and I am experiencing a similar problem. I did the non-proxy configuration:

> 'oidc_login_proxy_ldap' => false

 but I see the same DN not found message for user {samaccountname}. This attribute is the uid that is used for LDAP authentication and is working normally.
These are the attributes of the TOKEN according to the README:

> 'oidc_login_attributes' => array (
>    'id' => 'preferred_username',
>    'mail' => 'email',
>)

Am I missing an attribute? I did an inspection of all methods that are called to ldap_user and I confess that I got lost in method calls. There are many until reaching the 356 line of the User_LDAP.php class

> // getting dn, if false the user does not exist. If dn, he may be mapped only, requires more checking.
>   $ user = $ this-> access-> userManager-> get ($ uid);

I'm sorry for my bad english.

**Comment by pulsejet** [comment]
From the readme
> The `id` attribute in `oidc_login_attributes` must return the "Internal Username" (see expert settings in LDAP integration)

@andreisilvasantos is this correctly configured? Giving the login username as `id` might not work.

**Comment by andreisilvasantos** [comment]
Thanks for the tip. I reviewed the algorithm and had to make the same change as the colleague above:

> $uid = strtoupper($profile[$attr['id']]);

As the nextcloud imported the users creating the 'Internal User' using the Object UID in upper case and the keycloak imports in lower case, it is necessary to make this conversion on one side. As I have little knowledge of java and the keycloak architecture I chose to make this adjustment in the oidc plugin.
